torchao/prototype/low_bit_optim/subclass_4bit.py

```diff
@@ -0,0 +1,164 @@
+import math
+
+import torch
+from torch import Tensor
+from torchao.dtypes.utils import _implements, _ATEN_OP_OR_TORCH_FN_TABLE
+
+from .subclass_8bit import create_dynamic_map
```

Comment:
wanna pull this out into its own file?

Reply:
It's only 1 function so I personally think keeping it like this is fine. But if you prefer moving it to a separate file (to avoid dependency of "same-level" files), I'm ok too.